keps/sig-node/3857-rro-mounts/README.md

```diff
@@ -0,0 +1,1085 @@
+### Graduation Criteria
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, [feature gate] graduations, or as
+something else. The KEP should keep this high-level with a focus on what
+signals will be looked at to determine graduation.
+
+Consider the following in developing the graduation criteria for this enhancement:
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+- [Feature gate][feature gate] lifecycle
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning)
+or by redefining what graduation means.
+
+In general we try to use the same stages (alpha, beta, GA), regardless of how the
+functionality is accessed.
+
+[feature gate]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
```

Comment:
Updated

```
Upgrade: No action is needed. Existing readonly mounts will remain non-recursively readonly.

Downgrade:
- On downgrading kube-apiserver, the `[]volumeMounts.recursiveReadOnly` properties must be
  unset by the user. Otherwise kube-apiserver will generate an "unknown field" error.
- On downgrading kubelet, the `[]volumeMounts.recursiveReadOnly` properties should be unset by the user.
  Otherwise kubelet will ignore them, or maybe generate an "unknown field" error,
  depending on its implementation.
  The user may check the existence of the `[]containerStatuses.[]volumeMount.recursiveReadOnly`
  property to see if it was ignored.
- On downgrading the CRI or OCI runtime, no further action is needed, unless the `RecursiveReadOnly` mode
  is set to `Enabled`.
  kubelet will raise an error if it is set to `Enabled`.
  `IfPossible` will be treated as `Disabled`.
  The user may check the value of the `[]containerStatuses.[]volumeMount.recursiveReadOnly`
  property to see how it was treated.
```

Reply:
Reworded
```
+- On downgrading kube-apiserver, the `[]volumeMounts.recursiveReadOnly` property will be lost
+  and will not be propagated to kubelet.
+  If the mode was set to non-`Disabled`, this will result in producing writable mounts.
+  It is the user's responsibility to use the correct version of kube-apiserver
+  when they need non-`Disabled` mode.
+
+- On downgrading kubelet, the `[]volumeMounts.recursiveReadOnly` properties will be lost,
+  and the `[]containerStatuses.[]volumeMount.recursiveReadOnly` status will not be updated.
+  It is the user's responsibility to use the correct version of kubelet when they need to check
+  `[]containerStatuses.[]volumeMount.recursiveReadOnly`.
+
+- On downgrading the CRI or OCI runtime, if the `RecursiveReadOnly` mode is set to `Enabled`,
+  kubelet will raise an error.
+  `IfPossible` will be just treated as `Disabled`.
```

>  you need to explain why this isn't actually a problem.

Let me say "It is the user's responsibility to use the correct version"